core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -154,13 +169,12 @@ boolean forceRefresh() {
       return false;
     }
     synchronized (instanceDataGuard) {
-      // If a scheduled refresh exists and can't be canceled, fail the operation
-      if (nextInstanceData != null && !nextInstanceData.cancel(false)) {
-        return false;
+      // If a scheduled refresh doesn't exist or hasn't started, perform one immediately
+      if (nextInstanceData == null || nextInstanceData.cancel(false)) {
+        nextInstanceData = Futures.immediateFuture(performRefresh());
```

Comment:
Oh, I like this!

Could you avoid the `blockOnNestedFuture` call in this case, too?

```java
if (nextInstanceData == null || nextInstanceData.cancel(false)) {
  currentInstanceData = performRefresh();
  nextInstanceData = immediateFuture(currentInstanceData);
} else {
  currentInstanceData = blockOnNestedFuture(nextInstanceData);
}
```

Now you have only one call to `blockOnNestedFuture`, because the other cases are simpler.

And since those two lines in the if block are the same as what you do in the constructor, you could have a simple `replaceCurrentInstanceData()` method that's called from both places, if you want.